athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchSchemaUtils.java

```diff
@@ -0,0 +1,252 @@
+/*-
+ * #%L
+ * athena-elasticsearch
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.elasticsearch;
+
+import com.amazonaws.athena.connector.lambda.data.SchemaBuilder;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class has interfaces used for the parsing and creation of a schema based on an index mapping retrieved
+ * from an Elasticsearch instance. It also has an interface for converting Elasticsearch data types to Apache Arrow.
+ */
+class ElasticsearchSchemaUtils
+{
+    private static final Logger logger = LoggerFactory.getLogger(ElasticsearchSchemaUtils.class);
+
+    // Used in parseMapping() to store the _meta structure (the mapping containing the fields that should be
+    // considered a list).
+    private LinkedHashMap<String, Object> meta = new LinkedHashMap<>();
```

Comment:
i dont think this needs to be a member variable, you can create it in parseMappings and pass it into inferSchema. That allows you to keep this class stateless and avoid confusing expectations around what order the methods in this class need to be called from.

Reply:
moved to parseMapping() and implemented as a passed argument to inferSchema()